packages/aws-cdk-lib/aws-stepfunctions/lib/states/custom-state.ts

```diff
@@ -68,10 +68,17 @@ export class CustomState extends State implements IChainable, INextable {
    * Returns the Amazon States Language object for this state
    */
   public toStateJson(): object {
-    return {
+    const state = {
       ...this.renderNextEnd(),
       ...this.stateJson,
       ...this.renderRetryCatch(),
     };
+
+    // merge the Retry filed defined in the stateJson into the state
```

Comment:
The Retry is configured in two ways(the `addRetry` method and defining it in the stateJson), so we need to merge them.